schema/elements.go

```diff
@@ -27,6 +29,15 @@ type Schema struct {
 
 	once sync.Once
 	m    map[string]TypeDef
+
+	// Once used to protect the initialization of `lock` field.
+	lockOnce sync.Once
+	// Lock which protects writes to resolvedTypes. Used as pointer so that
+	// schema may be used as a value type
+	lock *sync.Mutex
+	// Cached results of resolving type references to atoms. Only stores
+	// type references which require fields of Atom to be overriden.
+	resolvedTypes map[TypeRef]Atom
```

Comment:
Is `m` above ever invalidated? If so we should probably invalidate this at the same time.

Reply:
`m` is not ever reset/recomputed